src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/jpa/CourtEvent.kt

```diff
@@ -150,5 +150,16 @@ class CourtEvent(
     return id == other.id
   }
 
+  fun isLatestAppearance(): Boolean {
+    return courtCase?.let {
+      this == courtCase!!.courtEvents!!.sortedBy { event ->
```

Comment:
I wonder it needs the `!!` - I thought it should be smart casted to non-nullable because of the `?`

Reply:
I the !! is on the child courtEvents

Reply:
although feels like they shouldn't be nullable maybe - empty lists

Reply:
I also thought court case wouldn't need the !! but it did. I'll revisit on next PR